deploy/scripts/src/cri-kiwi/test.ts

```diff
@@ -770,3 +820,19 @@ function getClientID (r: Response): string {
   if (clientId !== null && typeof clientId === 'string') return clientId
   fail('Client ID not found')
 }
+
+function getCodeFromUrl (url: string): string {
+  const urlParts = url.split('?')
+  const queryParams = urlParts[1].split('&')
+  for (const param of queryParams) {
+    const [key, value] = param.split('=')
+    if (key === 'code' && value !== null && typeof value === 'string') return value
+  }
```

Comment:
```suggestion
  const code = url.match(/code=([^&]*)/)[1]
  if (code !== null) return code
```
We can avoid loops here with a simple regex match